src/cmakeProject.ts

```diff
@@ -974,7 +1008,7 @@ export class CMakeProject {
             // For multi-root, the "onDidSaveTextDocument" will be received once for each project folder.
             // To avoid misleading telemetry, consider the notification only for the active folder.
             // There is always one active folder in a workspace and never more than one.
-            if (isActiveFolder(this.folder)) {
+            if (isActiveFolder(this.rootFolder)) {
```

Comment:
I don't think you want the workspace folder here.  This block of code should only run on the active CMakeProject.